src/plugins/mailer/builder.ts

```diff
@@ -0,0 +1,184 @@
+   */
+  private plainText = '';
+  /**
+   * The HTML contents of the e-mail. Note that the interpolation in the subject translation string will be escaped for security reasons.
+   */
+  private htmlText = '';
+  /**
+   * The language of the e-mail.
+   */
+  private language: string = DEFAULT_LANG;
+
+  private translate: TFunction;
+
+  constructor({
+    subject,
+    lang,
+  }: {
+    subject: {
+      text: (typeof MAIL)[keyof typeof MAIL];
+      translationVariables?: { [key: string]: string };
+    };
+    lang?: string;
+  }) {
+    if (lang) {
+      this.language = lang;
+    }
+
+    i18next.changeLanguage(this.language);
+    this.translate = i18next.t;
+
+    this.subject = this.translate(subject.text, {
+      ...subject.translationVariables,
+      interpolation: { escapeValue: false },
+    });
+  }
+
+  public addText(
+    mailTextId: (typeof MAIL)[keyof typeof MAIL],
+    translationVariables?: { [key: string]: string },
+  ): MailBuilder {
+    const unescapedText = this.translate(mailTextId, {
+      ...translationVariables,
+      interpolation: { escapeValue: false },
+    });
+    this.plainText += `\n${unescapedText}\n`;
+
+    const escapedText = this.translate(mailTextId, translationVariables);
+    this.htmlText += this.buildText(escapedText);
+
+    return this;
+  }
+
+  public addButton(
+    mailButtonTextId: (typeof MAIL)[keyof typeof MAIL],
+    callToActionLink: string,
+    translationVariables?: { [key: string]: string },
+  ): MailBuilder {
+    this.plainText += `\n${callToActionLink}\n`;
+
+    this.htmlText += this.buildButton(
+      callToActionLink,
+      this.translate(mailButtonTextId, translationVariables),
+    );
+
+    return this;
+  }
+
+  /**
+   * This function add the user agreement usually placed at the end of the e-mail.
+   */
+  public addUserAgreement(
+    mailButtonTextId: (typeof MAIL)[keyof typeof MAIL],
+    translationVariables?: {
+      [key: string]: string;
+    },
+  ): MailBuilder {
+    const unescapedSignUpButtonText = this.translate(mailButtonTextId, {
+      ...translationVariables,
+      interpolation: { escapeValue: false },
+    });
+    const unescapedText = this.translate(MAIL.USER_AGREEMENTS_MAIL_TEXT, {
+      signUpButtonText: unescapedSignUpButtonText,
+      graaspLandingPageOrigin: GRAASP_LANDING_PAGE_ORIGIN,
+      interpolation: { escapeValue: false },
+    });
+    this.plainText += `\n${unescapedText}\n`;
+
+    const escapedText = this.translate(MAIL.USER_AGREEMENTS_MAIL_TEXT, {
+      signUpButtonText: this.translate(mailButtonTextId, translationVariables),
+      graaspLandingPageOrigin: GRAASP_LANDING_PAGE_ORIGIN,
+    });
+    this.htmlText += this.buildText(
+      escapedText,
+      // Add margin top of -15px to remove 15px margin bottom of the button.
+      { 'text-align': 'center', 'font-size': '10px', 'margin-top': '-15px' },
+    );
+
+    return this;
+  }
+
+  public addSignUpNotRequested(): MailBuilder {
```

Comment:
It confused me when I read the Builder invocations with this function. It is not exclusive to the signUp. Maybe rename it to something less signUp-related.
```suggestion
  public addEmailSubscriptionNotice(): MailBuilder {
```

Since the message is something like: `You are receiving this email because of your account on the Graasp platform.`

Reply:
I have changed the string name to better reflect what it actually does.